### PR TITLE
Daps example

### PR DESCRIPTION
**Pull request by darashev-daiteap**
Add file transfer push example with two connectors that are connecting to DAPS.

Tasks:
- [x] In **requirements** section add information about DNS, hostname, and IP addresses needed for the deployment.
- [x] Add wait time for the DNS of the just created container registry to resolve. This often times out currently, because it could take more time
- [x] Add a couple of sencences in the README about what the script **./deploy-services.sh** does
- [x] Automate the steps in README, so that parameters are extracted automatically from CURL responses and new requests are automatically generated
- [x] Add explaination of what each CURL requests does exactly
- [x] Add warning for number of S3 keys limit

### CODE REVIEW

**Review by akafazov — CHANGES_REQUESTED** [review]
Added tasks in the PR description which need to be done

**Comment by akafazov** [comment]
Installation failed:

```sh

│ Warning: Helm release "edc-ionos-s3-consumer" was created but has a failed status. Use the `helm` command to investigate the error, correct it, then run Terraform again.
│ 
│   with helm_release.edc-ionos-s3-consumer,
│   on main.tf line 73, in resource "helm_release" "edc-ionos-s3-consumer":
│   73: resource "helm_release" "edc-ionos-s3-consumer" {
│ 
╵
╷
│ Warning: Helm release "edc-ionos-s3-provider" was created but has a failed status. Use the `helm` command to investigate the error, correct it, then run Terraform again.
│ 
│   with helm_release.edc-ionos-s3-provider,
│   on main.tf line 167, in resource "helm_release" "edc-ionos-s3-provider":
│  167: resource "helm_release" "edc-ionos-s3-provider" {
│ 
╵
╷
│ Error: timed out waiting for the condition
│ 
│   with helm_release.edc-ionos-s3-consumer,
│   on main.tf line 73, in resource "helm_release" "edc-ionos-s3-consumer":
│   73: resource "helm_release" "edc-ionos-s3-consumer" {
│ 
╵
╷
│ Error: timed out waiting for the condition
│ 
│   with helm_release.edc-ionos-s3-provider,
│   on main.tf line 167, in resource "helm_release" "edc-ionos-s3-provider":
│  167: resource "helm_release" "edc-ionos-s3-provider" {
│ 
╵
✘-1 ~/src/edc-ionos-s3/example/file-transfer-push-daps/terraform [daps-example ↑·1|✚ 3…13] $ 
```

**Comment by darashev-daiteap** [comment]
> Installation failed:
> 
> ```shell
> │ Warning: Helm release "edc-ionos-s3-consumer" was created but has a failed status. Use the `helm` command to investigate the error, correct it, then run Terraform again.
> │ 
> │   with helm_release.edc-ionos-s3-consumer,
> │   on main.tf line 73, in resource "helm_release" "edc-ionos-s3-consumer":
> │   73: resource "helm_release" "edc-ionos-s3-consumer" {
> │ 
> ╵
> ╷
> │ Warning: Helm release "edc-ionos-s3-provider" was created but has a failed status. Use the `helm` command to investigate the error, correct it, then run Terraform again.
> │ 
> │   with helm_release.edc-ionos-s3-provider,
> │   on main.tf line 167, in resource "helm_release" "edc-ionos-s3-provider":
> │  167: resource "helm_release" "edc-ionos-s3-provider" {
> │ 
> ╵
> ╷
> │ Error: timed out waiting for the condition
> │ 
> │   with helm_release.edc-ionos-s3-consumer,
> │   on main.tf line 73, in resource "helm_release" "edc-ionos-s3-consumer":
> │   73: resource "helm_release" "edc-ionos-s3-consumer" {
> │ 
> ╵
> ╷
> │ Error: timed out waiting for the condition
> │ 
> │   with helm_release.edc-ionos-s3-provider,
> │   on main.tf line 167, in resource "helm_release" "edc-ionos-s3-provider":
> │  167: resource "helm_release" "edc-ionos-s3-provider" {
> │ 
> ╵
> ✘-1 ~/src/edc-ionos-s3/example/file-transfer-push-daps/terraform [daps-example ↑·1|✚ 3…13] $ 
> ```
Can you show the logs from the connector containers?

**Comment by akafazov** [comment]
> 
> Can you show the logs from the connector containers?

The cluster is already deleted

**Comment by darashev-daiteap** [comment]
>  Add wait time for the DNS of the just created container registry to resolve. This often times out currently, because it could take more time
 
 https://github.com/Digital-Ecosystems/edc-ionos-s3/blob/f5670f9aa5b0fc82d7460b493876e354c575cee6/example/file-transfer-push-daps/terraform/build-and-push-docker-images/main.tf#L70-L83

**Comment by akafazov** [comment]
> > Add wait time for the DNS of the just created container registry to resolve. This often times out currently, because it could take more time
> 
> https://github.com/Digital-Ecosystems/edc-ionos-s3/blob/f5670f9aa5b0fc82d7460b493876e354c575cee6/example/file-transfer-push-daps/terraform/build-and-push-docker-images/main.tf#L70-L83

I know, but this needs to be increased

File: example/file-transfer-push-daps/README.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+```bash
+# Set the provider and consumer addresses
+export PROVIDER_ADDRESS=$(kubectl get svc -n edc-ionos-s3-provider edc-ionos-s3-provider -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+export CONSUMER_ADDRESS=$(kubectl get svc -n edc-ionos-s3-consumer edc-ionos-s3-consumer -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+
+```
+
+
+We will have to call some URL's in order to transfer the file:
```
Comment (akafazov) [inline example/file-transfer-push-daps/README.md:65]: We need better explanation than this. What does the call do?

File: example/file-transfer-push-daps/README.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+```
+
+2) Contract negotiation
+
+Copy the bracket of the `policy` from the response of the first curl into this curl and execute it.
+
+```bash
+curl --location --request POST "http://$CONSUMER_ADDRESS:8182/api/v1/management/contractnegotiations" \
```
Comment (akafazov) [inline example/file-transfer-push-daps/README.md:83]: Explain what does the request do?

File: example/file-transfer-push-daps/README.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+You will have an answer like the following:
+```bash
+{"createdAt":1672280687517,"id":"f80e1c17-810c-4ed5-b066-b286c189bb92"}
+```
+
+3) Contact Agreement id
+
+Copy the value of the `id` from the response of the previous curl into this curl and execute it.
+```bash
+curl -X GET -H 'X-Api-Key: password' "http://$CONSUMER_ADDRESS:8182/api/v1/management/contractnegotiations/<ID>"
```
Comment (akafazov) [inline example/file-transfer-push-daps/README.md:107]: Same here

File: example/file-transfer-push-daps/README.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+You will have an answer like the following:
+```bash
+{"createdAt":1672280687517,"updatedAt":1672280688733,"contractAgreementId":"1:83fc5fb4-84a9-4764-beea-4ff5446f91a0","counterPartyAddress":"http://$PROVIDER_ADDRESS:8282/api/v1/ids/data","errorDetail":null,"id":"f80e1c17-810c-4ed5-b066-b286c189bb92","protocol":"ids-multipart","state":"CONFIRMED","type":"CONSUMER"}
+```
+
+4) File transfer
+
+Copy the value of the `contractAgreementId` from the response of the previous curl into this curl and execute it.
+```bash
+curl --location --request POST "http://$CONSUMER_ADDRESS:8182/api/v1/management/transferprocess" \
```
Comment (akafazov) [inline example/file-transfer-push-daps/README.md:118]: description...

**Review by akafazov — CHANGES_REQUESTED** [review]

**Pull request merged**